src/components/signup/actions.js

```diff
@@ -0,0 +1,33 @@
+import {
+    SET_USER, SET_ERRORS, CLEAR_ERRORS, LOADING_UI, SET_AUTHENTICATED, SET_UNAUTHENTICATED,
+    USER_SIGNUP
+} from './actionType';
+
+export const loadingUI = () => ({
+    type: LOADING_UI,
+});
+export const setUser = payload => ({
+    payload,
+    type: SET_USER,
+});
+export const setErrors = payload => ({
+    payload,
+    type: SET_ERRORS,
+});
+export const clearErrors = () => ({
+    type: CLEAR_ERRORS,
+});
+export const setAuthenticated = payload => ({
+    payload,
+    type: SET_AUTHENTICATED,
+});
+export const setUnAuthenticated = payload => ({
+    payload,
+    type: SET_UNAUTHENTICATED,
+});
+export const userSignUp = (payload, history) => ({
+    history,
```

Comment:
I don't know if there is any rule against passing in more than one payload as parameters, but I think it is a better practice to have `payload` take all the payloads ie `get` history from payload. Since `history` is also a payload it might just be a repetition.

Reply:
Only pass in a singular `payload` object with all the information that you need.

Reply:
Thanks for the feeback. This npm package has been most helpful. Thanks